src/screens/login.js

```diff
@@ -2,53 +2,55 @@ import React, { Component } from 'react';
 import styled from 'styled-components/native';
 
 import FullView from '../components/fullView';
-import TextInput from '../components/textInput';
+import TextBox from '../components/textBox';
 import Button from '../components/button';
-import { primary } from '../commons/color';
+import { H1 } from '../commons/fontSize';
 export default class Login extends Component {
-    constructor(props) {
-        super(props)
-        this.state = {
-            email: '',
-            password: '',
-        }
+    state = {
+        email: '',
+        password: '',
+        isDisabled: true
     }
 
-    render() {
-        if (this.props.authenticated) {
+    componentDidUpdate(prevProps) {
+        if (!prevProps.authenticated && this.props.authenticated) {
             this.props.navigation.navigate("Register")
         }
+    }
 
+    render() {
+        const { error } = this.props
         return (
             <StyledFullView>
                 <TitleContainer>
                     <Title>Login</Title>
                 </TitleContainer>
-                <InternalView>
-                    <TextInputContainer>
-                        <TextInput title="Email" value={ this.state.email } onChangeText={ this.onChangeEmail } />
-                        <TextInput title="Password" value={ this.state.password } onChangeText={ this.onChangePassword } />
-                    </TextInputContainer>
-                    <Button title='Login' onPress={ this.login } />
-                </InternalView>
-
-                <Title>{this.props.err}</Title>
-                <OverrideButton title='Override to go to register' onPress={ () => this.props.navigation.navigate("Register") } />
+                <TextBoxContainer>
+                    <StyledTextInput title="Email" type='emailAddress' value={this.state.email} onChangeText={this.onChangeEmail} error={error} />
+                    <StyledTextInput title="Password" type='password' value={this.state.password} onChangeText={this.onChangePassword} />
+                </TextBoxContainer>
+                <LoginButton title='Login' onPress={this.login} disabled={this.state.isDisabled} />
```

Comment:
Disabled button is a good idea (Y). No need to have unnecessary client side validation